controllers/storagecluster/cephblockpools.go

```diff
@@ -159,8 +159,14 @@ func (obj *ocsCephBlockPools) ensureCreated(r *StorageClusterReconciler, instanc
 			}
 
 			r.Log.Info("Restoring original CephBlockPool.", "CephBlockPool", klog.KRef(cephBlockPool.Namespace, cephBlockPool.Name))
-			existing.ObjectMeta.OwnerReferences = cephBlockPool.ObjectMeta.OwnerReferences
+
+			existingMirroringSpec := existing.Spec.Mirroring
 			existing.Spec = cephBlockPool.Spec
+
+			// If mirroring is not set on storageCluster instance, don't update it
+			if instance.Spec.Mirroring == nil {
+				existing.Spec.Mirroring = existingMirroringSpec
+			}
```

Comment:
this is a bit confusing, is this what you are trying to achieve?
```suggestion
			// If mirroring is not set on storageCluster instance, don't update it
			if instance.Spec.Mirroring == nil {
				cephBlockPool.Spec.Mirroring = nil
			}
			existing.Spec = cephBlockPool.Spec
```

Reply:
I want to allow the updation of the mirroring field on the cephBlockPool instance from other components if the mirroring field is not set on storageCluster CR.

Initially, I was updating each field under spec one by one, if a new field is added it might be missed here and lead to bugs. For that, I am creating a temporary copy of the mirroring field from cephblockpool CR and mirroring is not set on storageCluster CR I am setting the value to the temporary variable. So, it will not update the mirroring field on cephblockpool when we don't enable mirroring from storageCluster CR.

The suggestion you made, will update the mirroring field on cephblockpool to be nil if mirroring is not set on storageCluster CR